src/types/AST.js

```diff
@@ -42,6 +47,11 @@ interface NumberLiteral {
   value: number;
 }
 
+interface LongLiteral {
```

Comment:
@xtuc perhaps this should be named `LongNumberLiteral`?

i.e. the `LongNumber` interface is intended to capture a concept which is similar to JavaScripts built in `number` type.